docs/samples/deployments/spark/presidio_anonymize_blobs.py

```diff
@@ -0,0 +1,153 @@
+# Databricks notebook source
+# MAGIC %md
+# MAGIC # Anonymize PII Entities in text files
+# MAGIC
+# MAGIC <br>Using Presidio, anonymize PII content of files in an Azure Storage account.
+# MAGIC
+# MAGIC <br>The following code sample will:
+# MAGIC <ol>
+# MAGIC <li>Import the content of text files located in an Azure Storage blob folder</li> # noqa D501
+# MAGIC <li>Anonymize the content using Presidio</li>
+# MAGIC <li>Write the anonymized content back to the Azure Storage blob account</li>
+# MAGIC </ol>
+
+# COMMAND ----------
+
+from azure.storage.blob import BlobServiceClient
+from presidio_analyzer import AnalyzerEngine
+from presidio_anonymizer import AnonymizerEngine
+from presidio_anonymizer.entities import AnonymizerConfig
+from pyspark.sql.types import StringType
+from pyspark.sql.functions import input_file_name, regexp_replace
+from pyspark.sql.functions import col, pandas_udf
+import pandas as pd
+
+dbutils.widgets.text(
+    "storage_account_name", "", "Blob Storage Account Name"
+)  # noqa: F821
+dbutils.widgets.text("storage_container_name", "", "Blob Container Name")  # noqa: F821
+dbutils.widgets.text(
+    "storage_account_access_key", "", "Storage Account Access Key"
+)  # noqa: F821
+dbutils.widgets.text(
+    "storage_input_folder", "input", "Storage Account Access Key"
+)  # noqa: F821
+dbutils.widgets.text(
+    "storage_output_folder", "output", "Storage Account Access Key"
+)  # noqa: F821
+
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC # Import the text files from Azure Blob storage
+
+# COMMAND ----------
+
+storage_account_name = dbutils.widgets.get("storage_account_name")  # noqa F821
+storage_container_name = dbutils.widgets.get("storage_container_name")  # noqa F821
+storage_account_access_key = dbutils.widgets.get(
+    "storage_account_access_key"
+)  # noqa F821
+
+
+blob_service_client = BlobServiceClient(
+    account_url="https://" + storage_account_name + ".blob.core.windows.net/",
+    credential=storage_account_access_key,
+)
+container_client = blob_service_client.get_container_client(storage_container_name)
+
+blob_names = container_client.list_blobs(
+    name_starts_with=dbutils.widgets.get("storage_input_folder") + "/"
```

Comment:
What if we instead mounted the blob storage to DBFS and then use `dbutils.fs.ls()` to obtain the list of blobs?

Reply:
works for reading, not for writing. commented in code

Reply:
That could be a permission issue on blob storage. What is the error that you got when writing?

Reply:
the thing is that i'm trying to write a text file per row, with the name of the file provided as a column in the df.
that is not supported ootb since only dataframes can be written to hdfs.
opening a file using native python always resulted in "file not found" error, and i could not get the worker nodes to know the hdfs. is is just me? do you have a reference to how writing text files within pandas udf (i.e. on worker nodes) can be done?

Reply:
> since only dataframes can be written to hdfs

Yes that is correct, hence I mentioned that for writing we can use the native Python API. The reason you might have received a "file not found" is because when using Python API you might need to use the path like this "/dbfs/..." instead of "wasbs:///" which you probably have in your filename column. This should help https://docs.databricks.com/data/databricks-file-system.html#access-files-on-mounted-object-storage

Reply:
Here is an example notebook that shows how to write a text file per row with the name of the file in a column. It mounts Blob Storage to DBFS then writes using the Python fs API https://github.com/syedhassaanahmed/databricks-notebooks/blob/master/df_row_to_multiple_files.py